src/index.ts

```diff
@@ -1,58 +1,235 @@
-const enum LogType {
-	DEBUG = "debug",
-	INFO = "info",
-	WARN = "warn",
-	ERROR = "error",
+export const enum LogType {
+  DEBUG = "debug",
+  INFO = "info",
+  WARN = "warn",
+  ERROR = "error",
+}
+
+export const enum Colors {
+  DARK_GRAY = "\x1b[48;5;240m",
+  CORN_FLOWER_BLUE = "\x1b[48;5;18m",
+  DARK_ORANGE = "\x1b[48;5;208m",
+  DARK_RED = "\x1b[48;5;88m",
+  BLACK = "\u001b[30m",
+  RED = "\u001b[31m",
+  GREEN = "\u001b[32m",
+  YELLOW = "\u001b[33m",
+  BLUE = "\u001b[34m",
+  MAGENTA = "\u001b[35m",
+  CYAN = "\u001b[36m",
+  WHITE = "\u001b[37m",
+  // add more if you wish
+}
+
+export const enum DateFormat {
+  YEAR_MONTH_DAY,
+  YEAR_DAY_MONTH,
+  DAY_MONTH_YEAR,
+  DAY_YEAR_MONTH,
+  MONTH_DAY_YEAR,
+  MONTH_YEAR_DAY,
 }
 
 const levelIndicators = {
-	debug: `🐛 `,
-	info: `ℹ️  `,
-	warn: `⚠️  `,
-	error: `⛔️ `,
+  debug: `🐛 `,
+  info: `ℹ️  `,
+  warn: `⚠️  `,
+  error: `⛔️ `,
 };
 
-const levelBackgroundColors = {
-	debug: ["\x1b[48;5;240m", "darkgray"],
-	info: ["\x1b[48;5;18m", "cornflowerblue"],
-	warn: ["\x1b[48;5;208m", "darkorange"],
-	error: ["\x1b[48;5;88m", "darkred"],
+type LeucineConfiguration = {
+  debugColor?: Colors;
+  infoColor?: Colors;
+  warnColor?: Colors;
+  errorColor?: Colors;
+  displayDate?: boolean;
+  dateFormat?: DateFormat;
+  displayTime?: boolean;
+  showMilliseconds?: boolean;
 };
 
-export const log = <T>(arg: T, level: LogType = LogType.DEBUG) => {
-	const backgroundColor = levelBackgroundColors[level];
-	const indicator = levelIndicators[level];
-
-	const fileName = import.meta.file ?? import.meta.url;
-	// @ts-ignore
-	if (typeof window === "undefined") {
-		console.log(
-			`\x1b[1m${backgroundColor[0]}${indicator} (${level}) in ${fileName}\x1b[0m\n\n`,
-			arg,
-			"\n",
-		);
-	} else {
-		console.log(
-			`%c${indicator} (${level})\n\n`,
-			`color:white;background-color:${backgroundColor[1]};`,
-			arg,
-			"\n\n",
-		);
-	}
-};
+class Leucine {
+  private debugColor = Colors.DARK_GRAY;
+  private infoColor = Colors.CORN_FLOWER_BLUE;
+  private warnColor = Colors.DARK_ORANGE;
+  private errorColor = Colors.DARK_RED;
 
-export const error = <T>(arg: T) => {
-	log(arg, LogType.ERROR);
-};
+  private displayDate = true;
+  private dateFormat = DateFormat.DAY_MONTH_YEAR;
+  private displayTime = true;
+  private showMilliseconds = false;
 
-export const debug = <T>(arg: T) => {
-	log(arg, LogType.DEBUG);
-};
+  constructor() {
+    this.log("Loaded Leucine!", LogType.INFO);
+  }
 
-export const info = <T>(arg: T) => {
-	log(arg, LogType.INFO);
-};
+  /**
+   * Recommended usage by power users only. For everyone else, use built in #set<type>() methods in builder pattern.
+   * @param {LeucineConfiguration} config
+   * @returns {this}
+   */
+  public configure(config: LeucineConfiguration): this {
```

Comment:
does `:this` get inferred in the return type here?

Reply:
It does get inferred yes, it can be removed